index.html

```diff
@@ -2945,6 +3042,168 @@ <h3>
+        </p>
+        <ol>
+          <li>Let <var>value</var> be <var>url</var>
+          </li>
+          <li>If <a>Type</a>(<var>value</var>) it not <a>String</a>, return
+          <code>undefined</code>.
+          </li>
+          <li>Otherwise, <a>parse</a> <var>value</var> using <var>manifest
+          URL</var> as the base URL and return the result.
+          </li>
+        </ol>
+      </section>
+      <section data-dfn-for="ShortcutItem" data-link-for="ShortcutItem">
+        <h3>
+          <dfn>params</dfn> member
+        </h3>
+        <p>
+          The <a>params</a> member of an <a>ShortcutItem</a> is a
+          <a>RequestParams</a> object defining the keys and values that should
+          be added to the <var>url</var> of a shortcut before navigation
+          occurs. See also: <a>launching a shortcut</a>.
+        </p>
+      </section>
+      <section data-dfn-for="ShortcutItem" data-link-for="ShortcutItem">
+        <h3>
+          <dfn>icons</dfn> member
+        </h3>
+        <p>
+          The <a>icons</a> member of an <a>ShortcutItem</a> member is an
+          <a>array</a> of <a>ImageResource</a>s that can serve as iconic
+          representations of the shortcut in various contexts.
+        </p>
+      </section>
+      <section>
+        <h3>
+          <dfn>Launching a shortcut</dfn>
+        </h3>
+        <p>
+          When <a>ShortcutItem</a> <var>shortcut</var> having
+          <a>WebAppManifest</a> <var>manifest</var> is invoked, run the
+          following steps:
+        </p>
+        <ol>
+          <li>Let <var>url</var> be a copy of <var>shortcut.url</var>.
+          </li>
+          <li>Let <var>header list</var> be a new empty <a data-xref-for=
+          "request">header list</a>.
+          </li>
+          <li>If <var>shortcut.params</var> <a data-cite="infra/#list-contain">
+            exists</a>:
+            <ol>
+              <li>Let <var>query</var> be the result of running the
+              <a>application/x-www-form-urlencoded</a> serializer with
+              <var>shortcut.params</var>
+              </li>
+              <li>Set <var>url</var>'s query component to <var>query</var>.
+              </li>
+            </ol>
+          </li>
+          <li>Append "Content-Type"/"application/x-www-form-urlencoded" to
+          <var>header list</var>.
+          </li>
+          <li>Let <var>body</var> be null.
+          </li>
+          <li>Let <var>browsing context</var> be the result of creating a new
+          <a>top-level browsing context</a>.
+          </li>
+          <li>
+            <a>Navigate</a> <var>browsing context</var> to a new <a>request</a>
+            whose url is <var>url</var>, header list is <var>header list</var>,
+            and body is <var>body</var>.
+          </li>
+        </ol>
+      </section>
+    </section>
+    <section>
+      <h2>
+        <dfn>RequestParams</dfn>
+      </h2>
+      <pre class="idl">
+        dictionary RequestParams {
```

Comment:
Why is this an empty dictionary? I guess you intend this to contain arbitrary key/values (not specifying which ones)? I don't think a dictionary is appropriate here. This should be an IDL [record](https://heycam.github.io/webidl/#idl-record).

However, I don't think we want RequestParams at all --- see comment above.

Reply:
Ah cool, I was unaware of `record`. I figured there had to be a similar generic, but had not found it yet. Thanks!